source/crud.txt

```diff
@@ -0,0 +1,94 @@
+=====================
+CRUD Operations Guide
+=====================
+
+.. default-domain:: mongodb
+
+.. toctree::
+   :caption: CRUD Operations
+   :hidden:
+
+   /crud/read-operations
+   /crud/write-operations
+   /crud/query-document
+   /crud/returned-data
+   /crud/compound-operations
+
+Overview
+--------
+
+MongoDB helps you store data persistently across different instances of
+your applications. CRUD (Create, Read, Update, Delete) operations allow
+you to work with the data stored inside of MongoDB. CRUD operations fall
+into two categories: read operations and write operations. All operation
+methods can be configured with a special ``options`` parameter.
+
+Read Operations
+---------------
+
+CRUD read operations help you get data out of MongoDB.
+Read operations provide flexibility so you can use them to
+view data in many different ways. While some read operations focus on
+allowing you to view data that exists in the database, other operations
+let you monitor the database for new data that matches certain criteria,
+and still other read operations let you apply a series of
+transformations and filters to data before returning a resulting
+emphemeral collection.
+
+- If you want to view documents already in a collection, you can use
+  :doc:`find() </usage-examples/find>` and
+  :doc:`findOne() </usage-examples/findOne>`. These methods accept a
+  query document that describes the documents you would like to view.
+  While ``findOne()`` returns a single document, ``find()`` returns a
+  :doc:`cursor </crud/read-operations/cursor>` that you can use to
+  navigate matched documents.
+
+- If you want to transform materialized views of documents in a
+  collection, you can use ``aggregate()``. This method accepts a
+  pipeline of aggregation commands that are run in sequence.
```

Comment:
Agg is _way_ more than materialized views.tTose only arrived in the most recent version (4.2) with $merge.

- [Aggregation](https://en.wikipedia.org/wiki/Aggregate_function) fundamentally means functions that create summary data for groups of rows/documents (e.g. sum, average, maximum, range, etc.). 
  - That definition is a little loose because you don't need to use a $group or otherwise group documents (though at that point why are you using agg?)
- MongoDB also lets you filter and manipulate your working set. (e.g. $filter, $project, $addFields, etc.)
- You define MongoDB aggregations in a pipeline of stages that each operate on the result set of the previous stage in the pipeline

Reply:
I think we should circle back to this. Summing up the aggregation framework in a couple of sentences is a nontrivial task... unfortunately I don't think we have a concrete replacement for my too-simplified explanation right now. Going to reword it and see if we can iterate from that.